lib/graphql/types/string.rb

```diff
@@ -7,7 +7,13 @@ class String < GraphQL::Schema::Scalar
 
       def self.coerce_result(value, ctx)
         str = value.to_s
-        str.encoding == Encoding::UTF_8 ? str : str.encode(Encoding::UTF_8)
+        if str.encoding == Encoding::UTF_8
+          str
+        elsif str.frozen?
+          str.encode(Encoding::UTF_8)
+        else
+          str.encode!(Encoding::UTF_8)
+        end
```

Comment:
This _could_ technically break code if people pass strings which are: 

- mutable 
- non-utf-8 
- expected to be reused later 

🤞